src/bp/core/services/telemetry/legacy-stats.ts

```diff
@@ -0,0 +1,186 @@
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.ConfigProvider) private config: ConfigProvider,
+    @inject(TYPES.CMSService) private cmsService: CMSService,
+    @inject(TYPES.AuthService) private authService: AuthService,
+    @inject(TYPES.UserRepository) private userRepository: UserRepository,
+    @inject(TYPES.BotService) private botService: BotService,
+    @inject(TYPES.WorkspaceService) private workspaceService: WorkspaceService
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = 'https://telemetry.botpress.io/ingest'
+    this.lock = 'botpress:telemetry-legacyStats'
+  }
+
+  protected async getStats() {
+    const config = await this.config.getBotpressConfig()
+
+    return {
+      schema: '1.0.0',
+      timestamp: new Date().toISOString(),
+      uuid: uuid.v4(),
+      server: await this.getServerStats(),
+      license: {
+        type: process.IS_PRO_ENABLED ? 'pro' : 'ce',
+        status: await this.getLicenseStatus(),
+        isProAvailable: process.IS_PRO_AVAILABLE,
+        showPoweredBy: config.showPoweredBy
+      },
+      bots: {
+        count: await this.getBotsCount()
+      },
+      workspaces: {
+        count: await this.getWorkspacesCount(),
+        pipelines: {
+          stages: {
+            count: await this.getPipelineStagesCount()
+          }
+        }
+      },
+      flows: {
+        count: await this.getFlowCount()
+      },
+      nlu: {
+        intents: {
+          count: await this.getIntentsCount()
+        },
+        entities: {
+          count: await this.getEntitiesCount()
+        }
+      },
+      qna: {
+        count: await this.getQnaCount()
+      },
+      hooks: {
+        global: {
+          count: await this.getGlobalHooksCount()
+        }
+      },
+      actions: {
+        global: {
+          count: await this.getGlobalActionsCount()
+        },
+        bot: {
+          count: await this.getBotActionsCount()
+        }
+      },
+      contentElements: {
+        count: await this.cmsService.countContentElements()
+      },
+      users: {
+        superAdmins: {
+          count: config.superAdmins.length
+        },
+        collaborators: {
+          count: await this.getCollaboratorsCount()
+        },
+        chat: {
+          count: await this.userRepository.getUserCount()
+        }
+      },
+      auth: {
+        strategies: {
+          count: Object.keys(config.authStrategies).length
+        }
+      }
+    }
+  }
+
+  protected async getServerStats() {
+    return {
+      externalUrl: process.EXTERNAL_URL,
+      botpressVersion: process.BOTPRESS_VERSION,
+      clusterEnabled: yn(process.CLUSTER_ENABLED, { default: false }),
+      os: process.platform,
+      bpfsStorage: process.BPFS_STORAGE,
+      dbType: this.database.knex.isLite ? 'sqlite' : 'postgres',
```

Comment:
I think this is also in process.env.DATABASE ... anyways it does the same job

Reply:
Actually `process.env.DATABASE` was deprecated, we only handle `DATABASE_URL` and we extract the type from there. I think isLite is the only way to determine the database type at the moment